src/FixedPointDecimals.jl

```diff
@@ -454,6 +454,78 @@ function div_with_overflow(x::FD{T,f}, y::FD{T,f}) where {T<:Integer,f}
     return (reinterpret(FD{T,f}, v), b)
 end
 
+# Does not exist in Base.Checked, so just exists in this package.
+@doc """
+    FixedPointDecimals.fld_with_overflow(x::FD, y::FD)::Tuple{FD,Bool}
+
+Calculates the largest integer less than or equal to `x / y`, checking for overflow errors
+where applicable, returning the result and a boolean indicating whether overflow occured.
+Throws a DivideError on divide-by-zero.
+
+The overflow protection may impose a perceptible performance penalty.
+
+See also:
+- `Base.checked_fld`.
+"""
+function fld_with_overflow(x::FD{T,f}, y::FD{T,f}) where {T<:Integer,f}
+    C = coefficient(FD{T, f})
+    # This case will break the fld call below. This can only happen when f is 0 as y.i
+    # cannot be -1 otherwise.
+    if T <: Signed && x.i == typemin(T) && y.i == -1
```

Comment:
Can you add the `f == 0` check here too? I don't know if julia would be able to infer that away?
```suggestion
    if T <: Signed && f == 0 && x.i == typemin(T) && y.i == -1
```

Reply:
I don't know what I was thinking but that comment was wrong. I guess I was looking at the integral part of `y`. `y.i` can be -1 when you have e.g. `FixedDecimal{Int8,2}(-0.01}`. Apologies.

Reply:
aha, right! 👍 

Then we should make sure to test those cases too! Thanks for looking through that!